app/models/reservation_status.rb

```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+class ReservationStatus < ApplicationRecord
+  has_many :reservations, dependent: :destroy
```

Comment:
This `dependent` option here might be quite dangerous in the future. Imagine a situation that you have 1000 reservations and you have status `approved` for instance. Now you get the requirement that status `approved` is not valid anymore and we changing this status to `paid` for instance. 

Having this option may delete all the reservations with particular status in the database. Generally, I know that you will remember about it and you will not let happen that kind of situation, but remember that in client projects you will work with multiple developers over the months, and having that kind of `dependent` option is more dangerous than not having it. Remember that people making mistakes and people often forgetting about something, so to minimize the chance of data loss we should avoid that kind of logic.

Btw. it's just free considerations, here it's not that important to change this, but in future, you should keep in mind that it's quite dangerous.

Reply:
thank you for pointing this out, I haven't thought of that before. 🙈  I guess I will take Wojtek's above advice and remove the table reservation_statuses and try to introduce module instead. Let's see how this works out.